Scripts/PAIF-N/01-deploy-vcf-workload-domain.ps1

```diff
@@ -0,0 +1,345 @@
+<#
+# © 2024 Broadcom.  All Rights Reserved.  Broadcom Confidential.  The term "Broadcom" refers to
+# Broadcom Inc. and/or its subsidiaries.
+#>
+
+<#
+.SYNOPSIS
+
+This script deploys a VI workload domain
+
+.DESCRIPTION
+
+This script creates a VI workload domain using the PowerCLI SDK module for VCF SDDC Manager.
+
+The script can be broken into two main parts. First, the ESXi hosts are being commissioned.
+Then, the actual VI domain is created using the commissioned ESXi hosts.
+
+Both steps - ESXi host commissioning and VI domain creations - are three-stage operations themselves.
+The commissioning/creation specs are constructed. The specs are validated. The actual operation is invoked.
+The validation and operation invocation are long-running tasks. This requires awaiting and status tracking
+until their completion. The waiting for validation and the actual operation is done using helper cmdlets -
+Wait-VcfValidation and Wait-VcfTask, located in utils sub-folder.
+
+On completion a new VI workload domain reflecting the given parameters should be created.
+
+.NOTES
+
+Prerequisites:
+ - A VCF Management Domain
+ - A minimum of three free hosts marked with the appropriate storage
+
+"Global parameters", "Host commissioning parameters", "Workload domain creation parameters" should be updated to
+reflect the environment they are run in. This may require altering the spec creation script.
+
+#>
+
+$ErrorActionPreference = 'Stop'
+$SCRIPTROOT = ($PWD.ProviderPath, $PSScriptRoot)[!!$PSScriptRoot]
+. (Join-Path $SCRIPTROOT 'utils/Wait-VcfTask.ps1')
+. (Join-Path $SCRIPTROOT 'utils/Wait-VcfValidation.ps1')
+
+
+# Configure PowerCLI
+Set-PowerCLIConfiguration -Scope User -ParticipateInCEIP $false -Confirm:$false
+Set-PowerCLIConfiguration -scope user -InvalidCertificateAction Ignore -Confirm:$false
```

Comment:
Drop these settings as their values should be the user's own settings. If you feel the user would not find how to change these, put a comment instead.